src/cli.zig

```diff
@@ -323,6 +341,35 @@ test "parse_size" {
     try expectEqual(@as(u64, 1000 * kib), parse_size("  1000  kb "));
 }
 
+/// Given a limit like `10GiB`, return the maximum power-of-two count of `T`s
+/// that can fit in the limit.
+fn parse_size_to_count(comptime T: type, string_opt: ?[]const u8, comptime default: u32) u32 {
+    comptime assert(default >= 2048);
+    comptime assert(math.isPowerOfTwo(default));
+
+    var result: u32 = undefined;
+    if (string_opt) |string| {
+        const byte_size = parse_size(string);
+        const count_u64 = math.floorPowerOfTwo(u64, @divFloor(byte_size, @sizeOf(T)));
+        const count = math.cast(u32, count_u64) catch |err| switch (err) {
+            error.Overflow => fatal("size value is too large: '{s}'", .{string}),
+        };
+        if (count < 2048) fatal("size value is too small: '{s}'", .{string});
+        assert(count * @sizeOf(T) <= byte_size);
+
+        result = count;
+    } else {
+        result = default;
+    }
+
+    // SetAssociativeCache requires a power-of-two cardinality and a minimal
+    // size.
+    assert(result >= 2048);
+    assert(math.isPowerOfTwo(result));
```

Comment:
Style-wise I try to avoid `undefined` when it isn't too complicated to do so, since the type system doesn't check it. For example:

```suggestion
    var result: ?u32 = null;
    if (string_opt) |string| {
        const byte_size = parse_size(string);
        const count_u64 = math.floorPowerOfTwo(u64, @divFloor(byte_size, @sizeOf(T)));
        const count = math.cast(u32, count_u64) catch |err| switch (err) {
            error.Overflow => fatal("size value is too large: '{s}'", .{string}),
        };
        if (count < 2048) fatal("size value is too small: '{s}'", .{string});
        assert(count * @sizeOf(T) <= byte_size);

        result = count;
        // SetAssociativeCache requires a power-of-two cardinality and a minimal
        // size.
        assert(result >= 2048);
        assert(math.isPowerOfTwo(result));
    }
    return result orelse default;
```

Reply:
Makes sense. How about the latest version? I'd love to keep asserts just before return to clearly signal a post-codition, and to double-check the defaults as well.